### PR TITLE
finish homework week3

### PR DESCRIPTION
**Pull request by baharehsichani**

### CODE REVIEW

**Review by aina21 — CHANGES_REQUESTED** [review]
remove unrelated file for week2 from this PR

File: javascript/javascript1/week3/index1.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+
+function calculateTime(arr){
+
+  const calculateHour=arr.destinationDistance / arr.speed
+  const calculateMinute=arr.destinationDistance % arr.speed
+   
+
+  var str=calculateHour.toString()
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:38]: don't use var. you can use const instead

File: javascript/javascript1/week3/index1.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  var numarray=str.split('.')
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:39]: const

**Review by aina21 — CHANGES_REQUESTED** [review]

File: javascript/javascript1/week3/index1.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  var a=new Array()
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:40]: const

File: javascript/javascript1/week3/index1.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+console.log("-------------------------------")
+
+
+
+//*****  Series duration of my life ******
+
+
+let age =80
+
+age*=(365*24*60)
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:57]: const age = (365*24*60);

File: javascript/javascript1/week3/index1.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+let seriesDurations = [
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:59]: const

File: javascript/javascript1/week3/index1.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    hours: 8,
+    minutes: 36,
+  },
+]
+
+function logOutSeriesText(){
+
+  for (let i=0 ; i< seriesDurations.length ; i++){
+
+    let dayToMinutes= seriesDurations[i].days*24*60
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:84]: const

File: javascript/javascript1/week3/index1.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    let hourToMinutes= seriesDurations[i].hours*60
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:85]: const

File: javascript/javascript1/week3/index1.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+    let totalMinutes= dayToMinutes+hourToMinutes+seriesDurations[i].minutes
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:87]: const totalMinutes= (dayToMinutes+hourToMinutes+seriesDurations[i].minutes) * 100;

File: javascript/javascript1/week3/index1.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    totalMinutes*=100
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:88]: then you don't need this line

File: javascript/javascript1/week3/index1.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    let spendTime=(totalMinutes / age).toFixed(3)
```
Comment (aina21) [inline javascript/javascript1/week3/index1.js:89]: const

File: javascript/javascript1/week3/index2.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
+
+let notes = [];
+
+function saveNote(content, id) {
+  // write some code here
+  
+    let obj={
```
Comment (aina21) [inline javascript/javascript1/week3/index2.js:7]: obj is not a good name, I think note or you can also write like 
```
notes.push({
        content:content,
        id:id
    })
```

Comment (aina21) [reply javascript/javascript1/week3/index2.js:7]: you can read more about naming here
[https://github.com/chunmeishui/hyf-homework/pull/url](url)

File: javascript/javascript1/week3/index2.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes);
+
+function getNote(id) {
+    // your code here
+    for (i=0;i <=notes.length;i++){
+        if(id != ""){
```
Comment (aina21) [inline javascript/javascript1/week3/index2.js:25]: id type is the number, and you check here if id is not a empty string, it always true.

File: javascript/javascript1/week3/index2.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+          if(notes[i].id===id)
+            return notes[i]
+            else
+              return (console.log("This id does not exist"))
+        }
+         
+        return (console.log("Id has to be a number"))
```
Comment (aina21) [inline javascript/javascript1/week3/index2.js:32]: it's better to use return string value  or remove return and just log a string.

File: javascript/javascript1/week3/index2.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+console.log(notes);
+
+function getNote(id) {
+    // your code here
+    for (i=0;i <=notes.length;i++){
+        if(id != ""){
+          if(notes[i].id===id)
+            return notes[i]
+            else
+              return (console.log("This id does not exist"))
```
Comment (aina21) [inline javascript/javascript1/week3/index2.js:29]: I think it's not working if you call getId(2), because you return inside a loop.

**Comment by baharehsichani** [comment]
Thanks for the feedback

**Pull request merged**